torchaudio/functional/functional.py

```diff
@@ -971,3 +975,34 @@ def spectral_centroid(
                            device=specgram.device).reshape((-1, 1))
     freq_dim = -2
     return (freqs * specgram).sum(dim=freq_dim) / specgram.sum(dim=freq_dim)
+
+
+def apply_codec(waveform, sample_rate, compression, format, channels_first=True) -> Tensor:
```

Comment:
1. Can you annotate the signature so that it is TorchScript compatible?
2. Can you reorder `compression` and `format`? `compression` depends on `format`.
3. Can you make compression parameter optional so that it defaults to `None` (and update the docstring)?
4. Can you decorate the function with `requires_module` so that this will be only available to torchaudio installation with C++ extension, as we discussed?

```python
@_mod_utils.requires_module('torchaudio._torchaudio')
def apply_codec(
    waveform: Tensor,
    sample_rate: int,
    format: str,
    compression: Optional[float] = None,
    channels_first: bool = True,
) -> Tensor:
```